src/builder/components/question/survey-questions/survey-question.component.tsx

```diff
@@ -0,0 +1,109 @@
+import { Icon, Input, Switch } from 'antd';
+import Search from 'antd/lib/input/Search';
+import React, { SFC } from 'react';
+import { SurveyItem, SurveyItemType } from '../../../../common/types/survey-item.type';
+
+interface SurveyQuestionTypeProps {
+    question: SurveyItem;
+    onChange: (surveyItem: Partial<SurveyItem>) => void;
+}
+
+export const SurveyQuestion: SFC<SurveyQuestionTypeProps> = ({
+    question,
+    onChange,
+}: SurveyQuestionTypeProps): JSX.Element => {
+    switch (question.type) {
+        case SurveyItemType.SingleChoice:
+            return <YesNoSurveyQuestion onChange={onChange} question={question} />;
+        case SurveyItemType.MultiChoice:
+            return <MultiOptionQuestion onChange={onChange} question={question} />;
+        case SurveyItemType.OpenQuestion:
+            return <OpenQuestion onChange={onChange} question={question} />;
+    }
+};
+
+const YesNoSurveyQuestion: SFC<SurveyQuestionTypeProps> = ({
+    question,
+    onChange,
+}: SurveyQuestionTypeProps): JSX.Element => (
+    <div className="options yesno">
+        <Input
+            className="singleInput question-input"
+            addonBefore="Option 1"
+            defaultValue={question.responses[0]}
+            onBlur={({ target }) => {
+                console.log(target.value);
+                onChange({
+                    responses: [question.responses[0], target.value],
+                });
+            }}
+            onPressEnter={({ target }) => {
+                // dunno why but linter yells at me here
+                console.log(target.value);
+                onChange({
+                    responses: [question.responses[0], target.value],
+                });
+            }}
```

Comment:
`onPressEnter: React.KeyboardEvent<HTMLInputElement> does not have a `target.value` property, apparently. 

Even though it does... Because I'm using it.

But my linter complains and it's making the build fail.

Any thoughts?

Reply:
I would go one of these ways:
- Accept `value` prop from above
- Make this a `PureComponent` that has the value in its own state. 

It's very easy to bind this (just have an `onChange=this.onValueChange` function). Then `onEnter` can pass the value from state/props

Either way, the current value should _definitely_ be kept up to date in code, right now we don't have any idea what the value is until we press enter or blur

Reply:
Do we debounce the redux persist on `onChange` or only persist `onPressEnter | onBlur`?

Reply:
I like debounced `onChange` but no strong feelings